### ISSUE TITLE
Sometimes unreachableBlock gets called when internet is turned back on.

### ISSUE THREAD

**Issue by jgervin** [tonymillion/Reachability#59]
I am having the same issue is this http://stackoverflow.com/questions/19367351/tony-million-reachability-says-unreachable-when-connected

I tried to fix by using a delay for the code called in the unreachable block, but then cancel all delayed method calls if reachableBlock is called.

This is huge problem!

**Comment by 0xc00010ff** [comment]
Seeing this also in simulator and multiple devices on iOS8. `unreachable` block is called and `isReachable is false`, yet my requests are succeeding.

**Comment by SoundBlaster** [comment]
To fix it you can use `-reachabilityForInternetConnection` instead `-reachabilityWithHostname`.

**Comment by wuf810** [comment]
Yep, I'm seeing this too. Turn off Wifi and unreachableBlock gets called. Turn wifi back on and both reachableBlock and unreachableBlock get called. Unfortunately unreachableBlock get's called after. Normally I set a property here to indicate no network so it's a problem although I could work round it. Be nice to have a fix though....

**Comment by sadiq81** [comment]
Having the same problem in simulator 8.3

**Comment by egold** [comment]
+1 to having this issue. +1 to changing it to `reachabilityForInternetConnection` instead of `reachabilityWithHostname`

**Comment by donileo** [comment]
+1 having this issue as well and it looks like this is a bug with SCNetworkReachability to me as various libraries that use it (AFNetworking,KSReachability, this one) have this same issue
